npm/private/npm_translate_lock_helpers.bzl

```diff
@@ -268,34 +268,29 @@ def _get_npm_imports(importers, packages, replace_packages, patched_dependencies
             "link_package": _link_package(root_package, import_path),
         }
         linked_packages = {}
-        for dep_package, dep_version in dependencies.items():
-            if dep_version.startswith("link:"):
+        for dep_package, dep_package_key in dependencies.items():
+            if dep_package_key.startswith("link:"):
                 continue
-            if dep_version.startswith("npm:"):
-                # special case for alias dependencies such as npm:alias-to@version
-                maybe_package = dep_version[4:]
-            elif dep_version not in packages:
-                maybe_package = utils.package_key(dep_package, dep_version)
+            if dep_package_key not in linked_packages:
+                linked_packages[dep_package_key] = [dep_package]
             else:
-                maybe_package = dep_version
-            if maybe_package not in linked_packages:
-                linked_packages[maybe_package] = [dep_package]
-            else:
-                linked_packages[maybe_package].append(dep_package)
+                linked_packages[dep_package_key].append(dep_package)
         links["packages"] = linked_packages
         importer_links[import_path] = links
 
     patches_used = []
     result = {}
-    for package_key, package_info in packages.items():
-        name = package_info.get("name")
-        version = package_info.get("version")
-        friendly_version = package_info.get("friendly_version")
-        deps = package_info.get("dependencies")
-        optional_deps = package_info.get("optional_dependencies")
-        optional = package_info.get("optional")
-        transitive_closure = package_info.get("transitive_closure")
-        resolution = package_info.get("resolution")
+    for snapshot_key, snapshot_info in snapshots.items():
+        package_key = snapshot_info["package"]
+        package_info = packages[package_key]
+        name = package_info["name"]
+        version = package_info["version"]
```

Comment:
If these were all switched to `["key"]` earlier I would have saved hours of my life 🤦

I might prefactor that in a separate PR...

Reply:
Prefactored and merged: https://github.com/aspect-build/rules_js/pull/2471